src/extractor/guidance/node_based_graph_walker.cpp

```diff
@@ -48,6 +51,7 @@ void LengthLimitedCoordinateAccumulator::update(const NodeID from_node,
     coordinates.insert(coordinates.end(), current_coordinates.begin(), current_coordinates.end());
 
     accumulated_length += length;
+    accumulated_length = std::min(accumulated_length, max_length);
```

Comment:
Hm not sure this should be done in the walker - the user knows the max length, what's probably more interesting is the actual accumulated distance, no? Otherwise we make decisions for the call site so to speak.